### ISSUE TITLE
Firefox start crashes in sway while checking if it's default browser or not.

### ISSUE THREAD

**Issue by Kazimazi** [NixOS/nixpkgs#90436]
**Describe the bug**
Starting Firefox with "Always check if Firefox is your default browser" checked in preferences crashes it possibly only in sway as I haven't tried with other environments.

**To Reproduce**
1. Check "Always check if Firefox is your default browser"
2. Start Firefox
3. Start again, it gives option to start it in safe mode.

**Additional context**
Problem occures in swaywm.
I tried both firefox-wayland and unstable.firefox-wayland, no luck.
I might need to open a new issue for it, but making firefox default browser setting doesn't do anything in sway.

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.
`these paths will be fetched (0.05 MiB download, 0.28 MiB unpacked):
  /nix/store/wqfv04xz2rpl2326ijghn1f2qgiyplb4-bash-interactive-4.4-p23-dev
copying path '/nix/store/wqfv04xz2rpl2326ijghn1f2qgiyplb4-bash-interactive-4.4-p23-dev' from 'https://cache.nixos.org'...
 - system: `"x86_64-linux"`
 - host os: `Linux 5.6.18, NixOS, 20.03.2157.db31e48c5c8 (Markhor)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.6`
 - channels(root): `"nixos-20.03.2157.db31e48c5c8, nixos-unstable-20.09pre228622.029a5de0839"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos``

**Comment by primeos** [comment]
> Starting Firefox with "Always check if Firefox is your default browser" checked in preferences crashes it possibly only in sway as I haven't tried with other environments.

We'd need some more details on why it crashes. AFAIK the Wayland support in Firefox is still considered experimental, so this might be an upstream bug in Firefox.

E.g. the Sway developers suggest the following ([ISSUE_TEMPLATE.md](https://github.com/swaywm/sway/blob/master/ISSUE_TEMPLATE.md)):
> Additionally, problems with Firefox are almost certainly Firefox bugs, not sway bugs. Start by submitting your issue to the Firefox Bugzilla and come back here only after they confirm otherwise.

> I might need to open a new issue for it, but making firefox default browser setting doesn't do anything in sway.

This might be a NixOS issue. I'm not sure how Firefox implements this, but if they don't use a library it could be due to a missing reference for `xdg-settings`, etc (that could e.g. be confirmed with `strace`).

**Comment by Kazimazi** [comment]
> > 
> 
> This might be a NixOS issue. I'm not sure how Firefox implements this, but if they don't use a library it could be due to a missing reference for `xdg-settings`, etc (that could e.g. be confirmed with `strace`).

I guess thats exactly it, same problem with Chromium. Opening an issue for it.
